Detectors/TRD/base/src/PadPlane.cxx

```diff
@@ -158,3 +162,59 @@ void PadPlane::setNrows(int n)
   }
   mNrows = n;
 };
+
+double PadPlane::getPadRow(double z) const
+{
+  double lengthCorr = mLengthIPad / mLengthOPad;
+
+  // calculate position based on inner pad length
+  double padrow = -z / mLengthIPad + mNrows / 2;
+
+  // correct row for outer pad rows
+  if (padrow <= 1.0) {
+    padrow = 1.0 - (1.0 - padrow) * lengthCorr;
+  }
+
+  if (padrow >= double(mNrows - 1)) {
+    padrow = double(mNrows - 1) + (padrow - double(mNrows - 1)) * lengthCorr;
+  }
+
+  // sanity check: is the padrow coordinate reasonable?
+  if (padrow < 0.0 || padrow > double(mNrows)) {
+    LOG(error) << "ERROR: hit with z=" << z << ", padrow " << padrow
+               << " outside of chamber";
+  }
```

Comment:
@tdietel How often did you see this error come up? I have not seen it so far in my testing. Is it still necessary to keep this check around and should we go ahead and use `assert` if so?

Reply:
Is this the place where we discussed the assert()?

I have not looked at this error, or how often it occurs. I understand it should never occur, and should be caught before this function is even called. So I would go with the assert at this point.

Reply:
@jolopezl Should this assert line be `assert((0.0 <= padrow <= double(mNrows)))` instead? Because the program will crash if the expression evaluates to false, right?